fontforge/python.c

```diff
@@ -2625,14 +2589,172 @@ static PySequenceMethods PyFFContour_Sequence = {
     PyFFContour_Concat,		/* concat */
     NULL,			/* repeat */
     PyFFContour_Index,		/* subscript */
-    PyFFContour_Slice,		/* slice */
+    NULL,			/* slice */
     PyFFContour_IndexAssign,	/* subscript assign */
-    PyFFContour_SliceAssign,	/* slice assign */
+    NULL,			/* slice assign */
     PyFFContour_Contains,	/* contains */
     PyFFContour_InPlaceConcat,	/* inplace_concat */
     NULL			/* inplace repeat */
 };
 
+static PyObject *PyFFContour_Sub( PyObject *self, PyObject *key ) {
+    PyFF_Contour *cont = (PyFF_Contour *) self;
+    PyFF_Contour *ret;
+    Py_ssize_t start, end, step, len;
+    int i;
+
+    if ( PyInt_Check(key)) {
+        return PyFFContour_Index(self, PyNumber_AsSsize_t(key, PyExc_IndexError));
+    }
+    if ( !PySlice_Check(key) ) { 
+	PyErr_Format(PyExc_IndexError, "Contour indexed by integer only");
+	return( NULL );
+    }
+#if PY_VERSION_HEX > 0x03060100
+    if (PySlice_Unpack(key, &start, &end, &step) < 0) {
+	return( NULL );
+    }
+    len = PySlice_AdjustIndices(cont->pt_cnt, &start, &end, step);
+#elif PY_VERSION_HEX > 0x03020000
+    if (PySlice_GetIndicesEx(key, cont->pt_cnt, &start, &end, &step, &len) < 0) {
+	return( NULL );
+    }
+#else
+    if (PySlice_GetIndicesEx((PySliceObject *) key, cont->pt_cnt, &start, &end, &step, &len) < 0) {
+	return( NULL );
+    }
+#endif
+    if ( !(step == 1 || step == -1) ) {
+	PyErr_Format(PyExc_IndexError, "Only supported steps are 1 and -1");
+	return( NULL );
+    }
+
+    ret = (PyFF_Contour *)PyFF_ContourType.tp_alloc(&PyFF_ContourType, 0);
+    ret->is_quadratic = cont->is_quadratic;
+    ret->closed = false;
+    ret->pt_max = ret->pt_cnt = len;
+    ret->points = PyMem_New(PyFF_Point *,ret->pt_max);
+
+    for ( i = 0; i < len; i++ ) {
+	ret->points[i] = cont->points[i*step + start];
+	Py_INCREF(ret->points[i]);
+    }
+    return( (PyObject *) ret );
+}
+
+// Caller responsible for incrementing reference count
+static void PyFFContour_CInsertPoint(PyFF_Contour *self, PyFF_Point *p, int pos) {
+	int i;
+
+	if ( pos<0 || pos>=self->pt_cnt-1 )
+		pos = self->pt_cnt-1;
+	if ( self->pt_cnt >= self->pt_max ) {
+		/* Messes with self->points */
+		PyMem_Resize(self->points,PyFF_Point *,self->pt_max += 10);
+	}
+	for ( i=self->pt_cnt-1; i>pos; --i )
+	self->points[i+1] = self->points[i];
+	self->points[pos+1] = p;
```

Comment:
This seems a bit surprising that it would insert the point at `pos` instead of `pos+1`

Reply:
Don't you mean that the other way around? 

I agree that `pos` would be better but the interface has been implemented and documented the other way.

Reply:
Hmm, you don't mean it the other way around. Yeah, this seems wrong. Which means my test must be wrong ...

Reply:
The existing documentation for `insertPont` says "Adds point to the contour. If the optional third argument is give, the line will be added after the pos'th point, otherwise it will be at the end of the contour." So (weirdly) if you specify 0 the point should end up in position 1, and if you specify 1 it should end up in position 2.

Say there are three points and the user specifies `1`. Then you want to move existing point 2 to position 3 and existing point 1 to position 2, and then assign to 2. So I think that `i>pos` is the right test for the loop, and `pos+1` the right location, after all.

Reply:
The two lines are consistent, at least. The last iteration of the loop runs with i = pos + 1, which vacates that spot for an insertion at pos + 1. The calling convention is still sort of absurd, but what can we do?